### PR TITLE
[dynamo][guards-cpp-refactor] Func defaults and kwdefaults accessor

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120726
* __->__ #121338
* #121327

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

File: torch/csrc/dynamo/guards.cpp [hunk 1]
```diff
@@ -2299,6 +2299,84 @@ class DictGetItemGuardAccessor : public GuardAccessor {
...
+ * Represents func.__defaults__ accessor.
+ */
+class FuncDefaultsGuardAccessor : public GuardAccessor {
+ public:
+  FuncDefaultsGuardAccessor(
+      RootGuardManager* root,
+      py::object name,
+      std::string source,
+      py::handle example_value)
+      : GuardAccessor(root, name, source, example_value) {}
```
Comment (Skylion007) [inline torch/csrc/dynamo/guards.cpp:2312]:
```suggestion
      : GuardAccessor(root, std::move(name), std::move(source), example_value) {}
```

Comment (Skylion007) [reply torch/csrc/dynamo/guards.cpp:2312]: Looking at the GuardAccessor ctor, really all of the subclasses should do it...

Comment (anijain2305) [reply torch/csrc/dynamo/guards.cpp:2312]: Yes, all other subclasses have the same issue. Let me handle it in a separate PR.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**